Add Feature #3: Save current files (and its states)

rhl-bthr opened the pull request:
**Feature added in this PR:** Save current files (and its states)

**Features:**
 1. Clicking on the Suspend Button will save the current state in `cedit.activity/tmp`
 2. Reopening the activity will resume it from the saved state and also delete files created while suspending it.


**Tested on:**
Sugar 0.112. packaged environment (rdesktop). Ubuntu 16.04


**Test cases used:** (to check function of suspend button)
 1. Made three new files and added some text in them. Didn't save any of them. Clicked on suspend
 2. Made five new files and saved two of them. Clicked on suspend
 3. Made four new files and saved two. Modified one of them again. Didn't save this time. Clicked on suspend.
 4. For 1 to 3, reopened the activtiy and clicked on suspend again.
 

**Note:** Function of Stop Button is not changed. This gives the user a choice to save the state or not

@cristian99garcia, kindly review

Commit:
Feature #3: Save current files (and its states)

Features added:
1. Added suspend button to save the current state.
2. Resumes from saved state when activity is reopened.

modified:   CEdit.py
new file:   icons/suspend.svg
modified:   toolbars.py
modified:   view.py

Commit:
Fixed gi version warnings

	modified:   globals.py

cristian99garcia commented:
I don't think we need a new button, this functionality should work with "Stop" tool button using [Sugar Activity 'save' function](https://github.com/sugarlabs/sugar-toolkit-gtk3/blob/master/src/sugar3/activity/activity.py#L910).
On the other hand, it does not highlight syntax of automatically open files (until I save them).

Commit:
Changes requested in #3

1. Implemented the feature of stop in suspend itself.
2. Fixed Highilighting. Now, when activities are reopened, text highlighiting works for files.

modified:   CEdit.py
modified:   toolbars.py
modified:   view.py

Pro-Panda commented:
@cristian99garcia, thank you for pointing the mistakes

Changes Made:
 - [x] Implemented the suspend feature in the Stop Button itself
 - [x] Syntax get highlighted automatically after reopening the files

Kindly review again.

cristian99garcia commented:
Thanks very much, you did really a good job here!
